libc-top-half/musl/src/string/strspn_simd.c

```diff
@@ -0,0 +1,175 @@
+#if defined(__wasm_simd128__) && defined(__wasilibc_simd_string)
+// Skip Clang 19 and Clang 20 which have a bug (llvm/llvm-project#146574)
+// which results in an ICE when inline assembly is used with a vector result.
+#if __clang_major__ != 19 && __clang_major__ != 20
+
+#include <stdint.h>
+#include <string.h>
+#include <wasm_simd128.h>
+
+#if !defined(__wasm_relaxed_simd__) || !defined(__RELAXED_FN_ATTRS)
+#define wasm_i8x16_relaxed_swizzle wasm_i8x16_swizzle
+#endif
+
+// SIMDized check which bytes are in a set (Geoff Langdale)
+// http://0x80.pl/notesen/2018-10-18-simd-byte-lookup.html
+
+typedef struct {
+  __u8x16 lo;
+  __u8x16 hi;
+} __wasm_v128_bitmap256_t;
+
+__attribute__((always_inline))
+static void __wasm_v128_setbit(__wasm_v128_bitmap256_t *bitmap, int i) {
```

Comment:
Why not the following?

```suggestion
static void __wasm_v128_setbit(__wasm_v128_bitmap256_t *bitmap, uint8_t i) {
```

Reply:
Sure. It moves the cast out to the loop, best not rely on implicit `char` to `uint8_t`.

Arguably, this bit would be more correct with `unsigned char` and `unsigned`.

Reply:
Done.